resources/Materials/TestSuite/stdlib/materials/material_node_discovery.mtlx

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<materialx version="1.38">
+  <nodedef name="ND_material_def_material_1_0" node="material_def" version="1.0" isdefaultversion="true">
+    <output name="out" type="material" value="" />
+  </nodedef>
+  <standard_surface name="top_level_shader" type="surfaceshader" nodedef="ND_standard_surface_surfaceshader" version="1.0.1" xpos="620" ypos="950" />
```

Comment:
What is the motivation for linking this node to a specific NodeDef in the standard libraries, rather than controlling the NodeDef matching through node name and version number?  This seems like an unusual pattern that we would want to discourage in production usage, since only the node name and version number is robust to future changes in the library, but maybe there's an advantage here that I'm missing.

Reply:
I generated this where the nodedef is always written out to improve search performance for nodedef association. Not needed here so removing.